src/httpclient.rs

```diff
@@ -105,33 +105,67 @@ impl Client {
         })
     }
 
+    fn make_url(&self, path: &str) -> String {
+        let mut url: &str = &format!("{}{}", self.base_url, path);
+        if path.starts_with("http") {
+            url = path;
+        }
```

Comment:
```suggestion
        let url: &str; 
        if path.starts_with("http") {
            url = path;
        } else {
            url = &format!("{}{}", self.base_url, path);
        }
```
I think this would work